image/image_linux.go

```diff
@@ -688,33 +595,248 @@ var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Opt
+		optSnaps = append(optSnaps, &optSnap)
+	}
+	return optSnaps
+}
+
+func manifestFromLocalSnaps(snaps localSnapRefs, opts *Options) (map[string]snap.Revision, error) {
+	// Build a map of snaps for the manifest file
+	seedManifest := make(map[string]snap.Revision)
+
+	// Check local snaps again, but now after InfoDerived has been called. InfoDerived
+	// fills out the snap revisions for the local snaps, and we need this to verify against
+	// expected revisions.
+	for sn := range snaps {
+		// Its a bit more tricky to deal with local snaps, as we only have that specific revision
+		// available. Therefore the revision in the local snap must be exactly the revision specified
+		// in the manifest. If it's not, we fail.
+		specifiedRevision := opts.Revisions[sn.Info.SnapName()]
+		if !specifiedRevision.Unset() && specifiedRevision != sn.Info.Revision {
+			return nil, fmt.Errorf("cannot use snap %s for image, unknown/local revision does not match the value specified by revisions file (%s != %s)",
+				sn.Path, sn.Info.Revision, specifiedRevision)
+		}
+		if !sn.Info.Revision.Unset() {
+			seedManifest[sn.Info.SnapName()] = sn.Info.Revision
+		}
+	}
+	return seedManifest, nil
+}
+
+func selectAssertionMaxFormats(tsto *tooling.ToolingStore, sysSn *seedwriter.SeedSnap) error {
+	if sysSn == nil {
+		// nothing to do
+		return nil
+	}
+	snapf, err := snapfile.Open(sysSn.Path)
+	if err != nil {
+		return err
+	}
+	// XXX take also kernel into account
+	// XXX warning logic
+	maxFormats, _, err := snap.SnapdAssertionMaxFormatsFromSnapFile(snapf)
+	if err != nil {
+		return err
+	}
+	tsto.SetAssertionMaxFormats(maxFormats)
+	return nil
+}
+
+var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Options) error {
+	s, err := newImageSeeder(tsto, model, opts)
+	if err != nil {
+		return err
+	}
+
+	// TODO: developer database in home or use snapd (but need
+	// a bit more API there, potential issues when crossing stores/series)
+	db, err := asserts.OpenDatabase(&asserts.DatabaseConfig{
+		Backstore: asserts.NewMemoryBackstore(),
+		Trusted:   trusted,
+	})
+	if err != nil {
+		return err
+	}
+	f, err := s.start(db, optionSnaps(opts))
+	if err != nil {
+		return err
+	}
+
+	// We need to use seedwriter.DeriveSideInfo earlier than
+	// we might possibly know the system and kernel snaps to
+	// know the correct assertion max format to use.
+	// Fetch assertions tentatively into a temporary database
+	// and later either copy them or fetch more appropriate ones.
+	tmpDb := db.WithStackedBackstore(asserts.NewMemoryBackstore())
+	tmpFetcher := seedwriter.MakeRefAssertsFetcher(func(save func(asserts.Assertion) error) asserts.Fetcher {
+		return tsto.AssertionFetcher(tmpDb, save)
+	})
+
+	localSnaps, err := s.deriveInfoForLocalSnaps(tmpFetcher, tmpDb)
+	if err != nil {
+		return err
+	}
+
+	// Build a map of snaps for the manifest file, but after
+	// InfoDerived has been called. InfoDerived fills out the snap
```

Comment:
Perhaps this and related comments should be updated? `InfoDerived` is now called inside `deriveInfoForLocalSnaps`, so its less obvious  than before what is referred to here. See also comments inside `manifestFromLocalSnaps`

Reply:
Good point! Updated docs for manifestFromLocalSnaps and it's usage.